apollo-router/src/metrics/filter.rs

```diff
@@ -177,17 +206,22 @@ impl InstrumentProvider for FilteredInstrumentProvider {
     }
 }
 
-impl MeterProvider for FilterMeterProvider {
+impl opentelemetry::metrics::MeterProvider for FilterMeterProvider {
     fn versioned_meter(
         &self,
         name: impl Into<Cow<'static, str>>,
         version: Option<impl Into<Cow<'static, str>>>,
         schema_url: Option<impl Into<Cow<'static, str>>>,
         attributes: Option<Vec<KeyValue>>,
     ) -> Meter {
-        let delegate = self
-            .delegate
-            .versioned_meter(name, version, schema_url, attributes);
+        let delegate = match &self.delegate {
+            MeterProvider::Regular(provider) => {
```

Comment:
Why not put it in `impl MeterProvider {` above?

Reply:
Nice catch: [36cf7a4](https://github.com/apollographql/router/pull/4460/commits/36cf7a42f5e41bb28761ed48fba04e79e5bd077e)